src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/split-pane-header/split-pane-header.component.ts

```diff
@@ -45,19 +128,61 @@ export class SplitPaneHeaderComponent implements OnChanges {
         return this.files[this.activeFileIndex].file;
     }
 
+    /**
+     * handles selection of file from dropdown, propagates change to fileslectionsubject component for lock sync
+     * @param file to be selected
+     * @param idx index of the file from the dropdown
+     */
     handleFileSelect(file: FileWithHasMatch, idx: number): void {
```

Comment:
You could combine `handleFileSelect` and `handleFileSelectWithoutPropagation` and add a parameter which handles if the functions should propagate changes or not.